lib/MySQL_Session.cpp

```diff
@@ -185,6 +255,40 @@ void MySQL_Session::writeout() {
 	proxy_debug(PROXY_DEBUG_NET,1,"Thread=%p, Session=%p -- Writeout Session %p\n" , this->thread, this, this);
 }
 
+bool MySQL_Session::handler_special_queries(PtrSize_t *pkt) {
+	if (pkt->size==SELECT_VERSION_COMMENT_LEN+5 && strncmp((char *)SELECT_VERSION_COMMENT,(char *)pkt->ptr+5,pkt->size-5)==0) {
+		PtrSize_t pkt_2;
+		pkt_2.size=PROXYSQL_VERSION_COMMENT_LEN;
+		pkt_2.ptr=l_alloc(pkt_2.size);
+		memcpy(pkt_2.ptr,PROXYSQL_VERSION_COMMENT,pkt_2.size);
+		status=WAITING_CLIENT_DATA;
+		client_myds->DSS=STATE_SLEEP;
+		client_myds->PSarrayOUT->add(pkt_2.ptr,pkt_2.size);
+		l_free(pkt->size,pkt->ptr);
+		return true;
+	}
+	if ( (pkt->size < 25) && (pkt->size > 15) && (strncasecmp((char *)"SET NAMES ",(char *)pkt->ptr+5,10)==0) ) {
```

Comment:
I perfectly agree.
Although right now it is just 2 "special cases" . If they will become more, this part needs to be reviewed.